packages/connector-google/src/index.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * The Implementation of OpenID Connect of Google Identity Platform.
+ * https://developers.google.com/identity/protocols/oauth2/openid-connect
+ */
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ConnectorMetadata,
+  ValidateConfig,
+  SocialConnector,
+  GetConnectorConfig,
+} from '@logto/connector-types';
+import { conditional, assert } from '@silverhand/essentials';
+import got, { RequestError as GotRequestError } from 'got';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+  googleConfigGuard,
+  GoogleConfig,
+  defaultMetadata,
+  defaultTimeout,
+} from './constant';
+
+export class GoogleConnector implements SocialConnector {
+  public metadata: ConnectorMetadata = defaultMetadata;
+
+  public readonly getConfig: GetConnectorConfig<GoogleConfig>;
+
+  constructor(getConnectorConfig: GetConnectorConfig<GoogleConfig>) {
+    this.getConfig = getConnectorConfig;
```

Comment:
this can be optimized, see https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

can create an refactor issue to track.

Reply:
[LOG-2324](https://linear.app/silverhand/issue/LOG-2324/refactor-connectors-constructor)